CMakeLists.txt

```diff
@@ -20,6 +20,13 @@ find_package(Boost REQUIRED system filesystem serialization)
 find_package(Eigen REQUIRED)
 find_package(OpenCV REQUIRED)
 
+if ( OpenCV_VERSION VERSION_GREATER 2.4.9 )
+ add_definitions("-DOPENCV3=1")
```

Comment:
that's what I do in other packages yes but I told you the right thing to do is simply to check for the existence of the EPOCH variable in the header and that's it: this is lighter that adding a pre-processor variable.